src/main/cpp/blaze.cc

```diff
@@ -1012,10 +1018,24 @@ static void EnsureCorrectRunningVersion(const StartupOptions &startup_options,
     std::unique_ptr<blaze_util::IFileMtime> mtime(
         blaze_util::CreateFileMtime());
     if (!mtime->SetToNow(blaze_util::Path(startup_options.install_base))) {
-      string err = GetLastErrorString();
-      BAZEL_DIE(blaze_exit_code::LOCAL_ENVIRONMENTAL_ERROR)
-          << "failed to set timestamp on '" << startup_options.install_base
-          << "': " << err;
+      // Ignore permissions errors (i.e. if the install base is not writable):
+      bool raise_error = true;
+
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__CYGWIN__)
+      // On POSIX platforms, `SetToNow` is backed by `utime(2)` which can
+      // return `EROFS`/`EPERM`:
+      if (errno == EROFS || errno == EPERM) { raise_error = false; }
+#elif defined(_WIN32)
+      // On Windows, `SetToNow` is backed by `CreateFileW` + `SetFileTime`:
+      if (::GetLastError() == ERROR_ACCESS_DENIED) { raise_error = false; }
```

Comment:
I am not sure whether this is correct; unfortunately I don't have access to a Windows machine at the moment.